gui/layers/_vectors_layer/model.py

```diff
@@ -0,0 +1,624 @@
+#!/usr/bin/env python
+# title           : properties.py
+# description     :class Vectors layer that defines properties
+# author          :bryant.chhun
+# date            :1/16/19
+# version         :0.0
+# usage           :
+# notes           :
+# python_version  :3.6
+
+from typing import Union
+
+import numpy as np
+
+from .._base_layer import Layer
+from .._register import add_to_viewer
+from gui._vispy.scene.visuals import Line as LinesNode
+from vispy.color import get_color_names
+
+from vispy.util.event import Event
+
+from .view import QtVectorsLayer
+from .view import QtVectorsControls
+
+
+@add_to_viewer
+class Vectors(Layer):
+    """
+    Vectors layer renders lines onto the image.
+    There are currently two data modes supported:
+        1) image-like vectors where every pixel defines its own vector
+        2) coordinate-like vectors where x-y position is not fixed to a grid
+    Supports ONLY 2d vector field
+
+    Properties
+    ----------
+    vectors : np.ndarray
+        array of shape (N,4) or (N, M , 2)
+
+    data : np.ndarray
+        ABC implementation.
+        returns vectors property setter/getter
+
+    _original_data : np.ndarray
+        Used by averaging and length adjustments.
+        Is set during layer construction.
+        Is NOT updated if data is adjusted by assigning the 'vectors' property
+
+    averaging : str
+        kernel over which to average from one of _kernel_dict
+        averaging.setter adjusts the underlying data
+        subscribe an observer by registering it with "averaging_bind_to"
+
+    width : int
+        width of the line in pixels
+
+    length : int or float
```

Comment:
Not obvious to me what this does in all examples. I'm looking at the `demo_vectors_coord-like.py` example and changing length doesn't seem to change anything. Can you explain?

Reply:
I enable length and averaging only for image-like data.  Averaging only makes sense for image-like data where (n,m) kernels must all contain some value.  Additionally, length and averaging adjust the underlying data, which i think makes most sense with the image-like data.

Reply:
In that case can you set the `averaging` and `length` ComboBoxes to be `disabled` if `self._data_type` is `'projection'`. This will cause them to be grayed out and non-interactive